exp/bufferpool/pool.go

```diff
@@ -0,0 +1,53 @@
+// Package bufferpool supports object pooling for byte buffers.
+package bufferpool
+
+import "sync"
+
+const (
+	nBuckets = 20
+)
+
+// A default global pool.
+var Default Pool
+
+// A pool of buffers, in variable sizes.
+type Pool struct {
+	buckets [nBuckets]sync.Pool
+}
+
+// Get a buffer of the given length. its capacity may be larger than the
+// requested size.
+func (p *Pool) Get(size int) []byte {
+
+	for i := 0; i < nBuckets; i++ {
+		capacity := 1 << i
```

Comment:
This might be getting a bit in the weeds, but geometrically increasing array size by a factor r=2 will typically result in [~40% of each array going unused](https://ece.uwaterloo.ca/~dwharder/aads/Algorithms/Array_resizing/).

The above link suggests a few alternatives.  I believe Go uses r = √2 for its built-in slices.  We can use a table of constant, pre-calculated values.

Reply:
Yeah, I thought of some of this, but figured do the simple version first. I futzed with starting points a bit and it didn't really affect the benchmark, so I decided to just keep the code simple. It means those first few buckets basically won't be used internally (minimum possible size is going to be 8, and real messages will always be bigger than that).

I think it's probably going to be worth it to reduce wasted space; I can explore that now, or as a separate PR depending on your preference.

Reply:
>minimum possible size is going to be 8

Won't it be 1, since `i := 0` initially and `1 << 0` is `1`?  Similarly, isn't `1 << 1` equal to `2`?

Re scaling factor, totally agree that we should keep it simple.  Let's kick that can down the road.

Reply:
What I meant was: capnp messages are allocated in units of 8 bytes, so it just won't ever use the buckets whose capacity is smaller than that. But yes, those buckets will still be there.